src/mainwindow.cpp

```diff
@@ -75,12 +76,46 @@ MainWindow::~MainWindow()
 }
 
 void MainWindow::toggleFullScreen() {
-    if (isFullScreen())
+    if (isFullScreen()) {
+        QApplication::instance()->removeEventFilter(this);
+        showTabAndTop();
         showNormal();
-    else
+    }
+    else {
+        QApplication::instance()->installEventFilter(this);
+        hideTabAndTop();
         showFullScreen();
+    }
+}
+
+void MainWindow::hideTabAndTop() {
+    getTabBar()->hide();
+    getTopWidget()->hide();
+}
+
+void MainWindow::showTabAndTop() {
+    getTabBar()->show();
+    getTopWidget()->show();
+}
+
+bool MainWindow::eventFilter(QObject* /*object*/, QEvent* event)
+{
+    if (event->type() == QEvent::MouseMove && isFullScreen())
+    {
+        const auto mouseEvent = static_cast<QMouseEvent*>(event);
+        const int tabRegion = getTabBar()->height() + getTopWidget()->height() + 30;
+        // We don't have to check for visibilty as calling hide() on a hidden widget, or show() on a non-hidden widget is a no-op
```

Comment:
Since you left out the word "cheap" I would like to clarify that keeping `show()`ing a visible widget on every mouse move in the bigger part of the screen is acceptable only if that operation has negligible performance overhead rather than just being [idempotent](https://en.wikipedia.org/wiki/Idempotence). An example to the contrary would be if `QWidget::show()` unconditionally enqueued a repaint event.

Reply:
> if that operation has negligible performance overhead rather than just being [idempotent](https://en.wikipedia.org/wiki/Idempotence).

Noted, thank you.

> An example to the contrary would be if QWidget::show() unconditionally enqueued a repaint event.

True, I checked into code and found out that there's a [check in setVisible](https://code.woboq.org/qt5/qtbase/src/widgets/kernel/qwidget.cpp.html#_ZN7QWidget10setVisibleEb) already. That should be fine to qualify it as a no-op I think?

Reply:
Yes.